backend/tests/amendCaseNoteController.test.ts

```diff
@@ -274,16 +259,81 @@ describe('Amendment case note', () => {
 
       await controller.post(req, res)
 
-      // @ts-expect-error ts-migrate(2339) FIXME: Property 'flash' does not exist on type '{ origina... Remove this comment to see the full error message
       expect(req.flash).toHaveBeenCalledWith('amendmentErrors', [
         {
           href: '#moreDetail',
           text: 'Amendments can no longer be made due to the maximum character limit being reached',
         },
       ])
 
-      // @ts-expect-error ts-migrate(2339) FIXME: Property 'redirect' does not exist on type '{}'.
-      expect(res.redirect).toHaveBeenCalledWith('http://localhost:3002/prisoner/case-notes/amend-case-note/1')
+      expect(res.redirect).toHaveBeenCalledWith('/prisoner/A12345/case-notes/amend-case-note/1')
+    })
+  })
+
+  describe('areYouSure()', () => {
+    it('should render the confirm page', async () => {
+      await controller.areYouSure(req, res)
+
+      expect(res.render).toBeCalledWith('caseNotes/addCaseNoteConfirm.njk', {
+        offenderDetails: {
+          name: 'Smith, Bob',
+          profileUrl: '/prisoner/A12345',
+        },
+        offenderNo: 'A12345',
+        homeUrl: '/prisoner/A12345/case-notes',
+        breadcrumbText: 'Add more details to case note',
+      })
+    })
+  })
+
+  describe('confirm()', () => {
+    it('should save the case note if confirmed', async () => {
+      req.session = { draftCaseNoteDetail: { moreDetail: 'hello' } }
+      req.body = { confirmed: 'Yes' }
+
+      await controller.confirm(req, res)
+
+      expect(caseNotesApi.amendCaseNote).toBeCalledWith(res.locals, 'A12345', 1, {
+        text: 'hello',
+      })
+      expect(res.redirect).toBeCalledWith('/prisoner/A12345/case-notes')
+    })
+
+    it('should redirect if case note save fails', async () => {
+      req.session = { draftCaseNoteDetail: { moreDetail: 'hello' } }
+      req.body = { confirmed: 'Yes' }
+      const error400 = makeError('response', {
+        status: 400,
+        body: {
+          userMessage: 'createCaseNote.caseNote.text: Value is too long: max length is 4000',
+          developerMessage: 'createCaseNote.caseNote.text: Value too long: max length is 4000',
+        },
+      })
+      caseNotesApi.amendCaseNote = jest.fn().mockRejectedValue(error400)
+
+      await controller.confirm(req, res)
+
+      expect(caseNotesApi.amendCaseNote).toBeCalledWith(res.locals, 'A12345', 1, {
+        text: 'hello',
+      })
+      expect(req.flash).toHaveBeenNthCalledWith(1, 'amendmentErrors', [
+        { href: '#moreDetail', text: (error400 as any).response.body.userMessage },
+      ])
+      expect(req.flash).toHaveBeenNthCalledWith(2, 'formValues', {
+        moreDetail: 'hello',
+      })
+    })
+
+    it('should redirect if user does not confirm', async () => {
+      req.body = {}
+      req.session = { draftCaseNoteDetail: { moreDetail: 'hello' } }
+
+      await controller.confirm(req, res)
+
+      expect(caseNotesApi.amendCaseNote).not.toHaveBeenCalled()
```

Comment:
is it worth checking res.redirect here too?

Reply:
agreed, realised that and have fixed in my next PR :grin: